### PR TITLE
Fixing multiple models deploy 

### PR DESCRIPTION
**Pull request by YuliaKrimerman**
Closing issue 2 from https://issues.redhat.com/browse/RHOAIENG-16540

## Description

When selecting multiple models on a multi model server it was throwing and error when trying to deploy a second model.  Found on kserve cluster.
Before 
image
After 
Screenshot 2024-12-10 at 2 11 07 PM

## How Has This Been Tested?

2 Scenarios
1. Any cluster - In Data Science projects go to models, choose Multi-model serving platform. Add a model server.Deploy a model. Try to deploy a 2nd mode. Make sure that no error appears. 
2. Cluster that has NIM enabled- we need to make sure that the InferenceService gets the right name when deploying a kserve model or a NIM model.

## Test Impact

Fixed the failing test

## Request review criteria:

Self checklist (all need to be checked):
- [X] The developer has manually tested the changes and verified that the changes work
- [X] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [X] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change.

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

File: frontend/src/api/k8s/__tests__/inferenceServices.spec.ts [hunk 1]
```diff
@@ -140,7 +140,7 @@ describe('assembleInferenceService', () => {
 
   it('should handle name and display name', async () => {
     const displayName = 'Llama model';
-    const resourceName = 'llama-model';
+    const resourceName = 'my-inference-service-test';
```
Comment (mturley) [inline frontend/src/api/k8s/__tests__/inferenceServices.spec.ts:143]: This test seems weird - the resourceName should be a k8s-safe version of the displayName, so `llama-model` makes sense, but below we shouldn't be passing it as the `servingRuntimeName`. In our calls to `mockInferenceServiceModalData` we should pass `name: displayName, k8sName: resourceName` and for this test we shouldn't need to pass a `servingRuntimeName` at all.

**Review by mturley — CHANGES_REQUESTED** [review]
Quick fix for the test here

File: frontend/src/api/k8s/__tests__/inferenceServices.spec.ts (hunk 1 above)
Comment (YuliaKrimerman) [reply frontend/src/api/k8s/__tests__/inferenceServices.spec.ts]: Fixed

**Review by mturley — APPROVED** [review]

**Review by mturley — APPROVED** [review]
LGTM, tested on kserve-pm (modelmesh and kserve) and on `ai-dev04.kni.syseng` (borrowed from the NIM team) for the NIM form changes.

**Comment by mturley** [comment]
FYI @jeff-phillips-18 , fixing some more InferenceService name stuff here that was caused by https://github.com/opendatahub-io/odh-dashboard/pull/3464. Sorry I overlooked these when talking through that PR with you.

**Pull request merged**